### ISSUE TITLE
Support highlighting of lambda params?

### ISSUE THREAD

**Issue by Qwerp-Derp** [faustinoaq/vscode-crystal-lang#8]
Support highlighting of this:

    [1, 2, 3].map { |a| a + 1 }

The `a` in `|a|` should be highlighted as a parameter.

**Comment by faustinoaq** [comment]
It depends of theme, my extension already have parameter support.

Currently I'm using
https://marketplace.visualstudio.com/items?itemName=faustinoaq.theme-olivine-comments
to
get good crystal highlighting.

[image: Inline image 1]
Notice the blue a in |a|

**Comment by faustinoaq** [comment]
![screenshot_20170729_040047](https://user-images.githubusercontent.com/3067335/28743578-ba81dce0-7412-11e7-812b-1335bf303418.png)

Notice the blue a in |a|

**Comment by Qwerp-Derp** [comment]
Shouldn't it be highlighted in the same way as a parameter, though?

**Comment by faustinoaq** [comment]
Unless I change [this line](https://github.com/faustinoaq/vscode-crystal-lang/blob/master/syntaxes/crystal.json#L1329) to Ruby, VSCode themes does't recognize the crystal delimiter.

```json
"begin": "(?<=\\{|do|\\{\\s|do\\s)(\\|)",
"captures": {
  "1": {
    "name": "punctuation.separator.variable.crystal"
  }
},
```

**Comment by faustinoaq** [comment]
@Qwerp-Derp I found the issue the Crystal syntax `.tmLanguage` don't have any rules to parameters with types. As far I can I changed the `endCapture` to get all parameters using:

```
"end": "\\)\\s*$|\\)\\s*:",
```

Also I found a lot of dirty code when I converted `.tmLanguage` to `.json`

I'll do a PR fixing this issue.

Internally related to #7

**Comment by faustinoaq** [comment]
Fixed on #9
